src/overlays/actors/ovl_Boss_Hakugin/z_boss_hakugin.c

```diff
@@ -46,342 +113,3727 @@ ActorInit Boss_Hakugin_InitVars = {
+                }
+            }
+        }
+    }
+}
+
+/**
+ * Updates the segments of the lightning attack that Goht can shoot at the player while it's waiting. Each segment's
+ * alpha is decreased by 20 for every frame; if the alpha is below 60, the lightning segment will no longer be able to
+ * hurt the player.
+ */
+void BossHakugin_UpdateLightningSegments(BossHakugin* this, PlayState* play) {
+    GohtLightningSegment* lightningSegment;
+    s16 rand;
+    s32 i;
+
+    for (i = 0; i < GOHT_LIGHTNING_SEGMENT_COUNT; i++) {
+        lightningSegment = &this->lightningSegments[i];
+
+        if (lightningSegment->collider.base.atFlags & AT_HIT) {
+            Player* player = GET_PLAYER(play);
+
+            this->lightningHitSomething = true;
+            Player_PlaySfx(player, NA_SE_EN_COMMON_E_BALL_HIT);
+            lightningSegment->collider.base.atFlags &= ~AT_HIT;
+        }
+
+        if (lightningSegment->alpha > 0) {
+            lightningSegment->alpha -= 20;
+
+            if (lightningSegment->alpha <= 0) {
+                lightningSegment->alpha = 0;
+            } else if (lightningSegment->alpha <= 255) {
+                rand = (s32)Rand_Next() >> 0x10;
+                lightningSegment->rot.z = rand;
+
+                if (lightningSegment->alpha > 60) {
+                    CollisionCheck_SetAT(play, &play->colChkCtx, &lightningSegment->collider.base);
+                }
+            }
+        }
+    }
+}
+
+/**
+ * Updates the electric balls that Goht can shoot at the player.
+ */
+void BossHakugin_UpdateElectricBalls(BossHakugin* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    CollisionPoly* poly = NULL;
+    Vec3f resultPos;
+    Vec3f posB;
+    Vec3f normal;
+    Vec3f targetRot;
+    Vec3f* firstBallPos;
+    Vec3f* ballPosIter;
+    s32 pad;
+    f32 angleStep;
+    s32 bgId;
+    s32 playedHitSfx = false;
+
+    firstBallPos = &this->electricBallPos[0];
+
+    if (this->electricBallState == GOHT_ELECTRIC_BALL_STATE_FLY_FORWARD) {
+        if (Math_StepToF(&this->electricBallSpeed, 50.0f, 3.5f)) {
+            this->electricBallState = GOHT_ELECTRIC_BALL_STATE_FLY_TOWARDS_PLAYER;
+        }
+    } else if (this->electricBallState == GOHT_ELECTRIC_BALL_STATE_FLY_TOWARDS_PLAYER) {
+        angleStep =
+            BINANG_TO_RAD(Math3D_Vec3fDistSq(firstBallPos, &player->actor.world.pos) * (1.0f / SQ(120.0f)) * 32.0f);
+        angleStep = CLAMP_MAX(angleStep, M_PI / 4.0f);
+
+        Math_Vec3f_Diff(&player->actor.world.pos, firstBallPos, &targetRot);
+
+        if (BossHakugin_Vec3fNormalize(&targetRot)) {
+            BossHakugin_StepVector(&this->electricBallRot, &targetRot, angleStep);
+        }
+    } else if (this->electricBallState != GOHT_ELECTRIC_BALL_STATE_FADE_OUT) {
+        return;
+    }
+
+    Audio_PlaySfx_AtPos(&this->electricBallSfxPos, NA_SE_EN_COMMON_E_BALL - SFX_FLAG);
+
+    if (this->electricBallCount < 10) {
+        this->electricBallCount++;
+    }
```

Comment:
Could you make a define `ELECTRIC_BALL_COUNT_MAX 10` or something similar? For use of the struct, here, and below?

Reply:
Good idea. I think I got it all, but let me know if I missed any usages that you were thinking of.